apps/quick-learn-backend/src/common/utils/helper.ts

```diff
@@ -47,4 +47,44 @@ export default class Helpers {
       ? sanitizedContent.substring(0, 250) + '...'
       : sanitizedContent;
   }
+
+  static extractImageUrlsFromHtml(
+    input: string | { [key: string]: unknown }[],
+    label?: string,
+    isHtmlString = true
+  ): string[] {
+    // Regular expression to match <img> tags and extract the src attribute
+    const regex = /<img[^>]+src="([^"]+)"/g;
+    const urls: string[] = [];
+  
+    if (isHtmlString) {
+      // Input is a single HTML string
+      let match: RegExpExecArray | null;
+  
+      // Execute regex to find all matches
+      while ((match = regex.exec(input as string)) !== null) {
+        // Add the URL found to the array
+        if (match[1]) {
+          urls.push(match[1]);
+        }
+      }
+    } else {
```

Comment:
If we're making this label an optional parameter, then we should handle this over here as well. There should not be data called item[undefined].